### PR TITLE
Ensure gemspec specifies libhoney >= 1.9

### PR DESCRIPTION
**Pull request by christineyen**
In order to ensure new users scoop up the performance benefits that come along with `libhoney 1.9+`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,5 +2,4 @@ source 'https://rubygems.org'
 
 gemspec
 
-# pull in newer libhoney just for testing, since we need TestClient#reset
-gem 'libhoney', '>= 1.5.1'
+gem 'libhoney', '>= 1.9'
```
Comment (samstokes) [inline Gemfile]: Actually we should just remove the `gem 'libhoney'` declaration from the `Gemfile` altogether (and ditto for `Gemfile.rails{3,4}.rb`). (_Users_ of libhoney are unaffected by the contents of the Gemfile, it only affects dev and CI.)

I put that version override in here a zillion years ago when the gemspec still specified a much older version of libhoney, in order to override the gemspec-specified version to a newer version required to run the tests. For a while now the gemspec version has been new enough anyway, so the override isn't doing anything any more and we should remove it.

File: Gemfile.rails3.rb [hunk 2]
```diff
@@ -2,8 +2,7 @@
 
 gemspec
 
-# pull in newer libhoney just for testing, since we need TestClient#reset
-gem 'libhoney', '>= 1.5.1'
+gem 'libhoney', '>= 1.9'
```
Comment (samstokes) [inline Gemfile.rails3.rb]: Ditto, we can remove this.

File: Gemfile.rails4.rb [hunk 3]
```diff
@@ -2,7 +2,6 @@
 
 gemspec
 
-# pull in newer libhoney just for testing, since we need TestClient#reset
-gem 'libhoney', '>= 1.5.1'
+gem 'libhoney', '>= 1.9'
```
Comment (samstokes) [inline Gemfile.rails4.rb]: Ditto, we can remove this.

**Review by samstokes — APPROVED** [review]
This is a good idea, but worth noting that anyone who installed or updated honeycomb-rails since we released libhoney 1.9 (in August) has almost certainly got libhoney 1.9 already. We're specifying a _minimum_ version in the gemspec, but `bundle install` will grab the _latest_ version of all specified gems that satisfy the constraints. ([Supporting evidence](https://ui-dogfood.honeycomb.io/prod/datasets/shepherd/result/j39WeCkszXG): only honeycomb-rails traffic from libhoney < 1.9 is from honeycomb-rails 0.7.2, which was released in June.)

Anyone who's currently on libhoney older than 1.9 will need to do a `bundle update libhoney` or `bundle update honeycomb-rails` (the latter [will currently update their version of Rails too](https://app.asana.com/0/526085303034087/787297217065163), which they probably didn't want to do). This change doesn't affect that operation either - with or without this change, either of those `bundle update` commands will install the latest version of libhoney.

So the only effect of this will be to _prevent_ anyone from running honeycomb-rails with libhoney 1.8.1 if that's what they're explicitly trying to do.

Still probably a change worth making from a cleanup and documentation perspective, it just probably won't have any customer-visible impact.

**Comment by christineyen** [comment]
> worth noting that anyone who installed or updated honeycomb-rails since we released libhoney 1.9 (in August) has almost certainly got libhoney 1.9 already.

ah, okay, so asking these new prospects seeing performance problems what version of libhoney they're running was more a stalling tactic than a real hypothesis? crafty 👍

**Pull request merged**